pyro/poutine/enum_messenger.py

```diff
@@ -71,7 +71,7 @@ def _tmc_diagonal_sample(msg: Message) -> torch.Tensor:
     # find batch dims that aren't plate dims
     batch_shape = [1] * len(dist.batch_shape)
     for f in msg["cond_indep_stack"]:
-        if f.vectorized:
+        if f.dim is not None:
```

Comment:
`f.vectorized` just check if `f.dim is not None` so I took the liberty to replace it here. Alternatively I can change it back as follows:
```py
if f.vectorized:
    assert f.dim is not None
```

Reply:
yeah I'd aim for the more verbose version because it makes intent clearer